### ISSUE TITLE
Dapp Wallet Bridge is Flaky

### ISSUE THREAD

**Issue by samsiegart** [Agoric/Agoric#6294]
## Describe the bug
We have some existing work items related to the dapp wallet bridge:
- https://github.com/Agoric/agoric-sdk/issues/6027
- https://github.com/Agoric/agoric-sdk/issues/6280

There's also other issues, like refreshing the wallet ui breaks the dapp and doesn't give any indication, requiring the user to refresh and reconnect the dapp.

The bridge signaling logic and state management with local storage likely just needs to be redesigned and cleaned up. We should investigate and design something that results in a better user experience overall.

**Comment by samsiegart** [comment]
# Design Proposal

## TL;DR
Make the dapp connect directly to Keplr to load all the purse data instead of getting it from the wallet bridge. Then we can remove all the CapTP stuff from the bridge and just turn it into a one-way mechanism for dapps to write offer proposals to the wallet without a two-way connection. This also gets us closer, both technically and UX-wise, to our end goal of getting rid of the bridge.

## Design Sketch
### Current Purpose of Wallet Bridge
(1) There isn't currently a good way to see all the offer data in Keplr before signing it (<insert link to sign-mode-textual issue here>?), so we lose most of the benefits of offer safety at the UX level. To fix this, we have dapps send messages to the wallet UI with descriptions of offers that they want the user to approve. The wallet UI is the single trusted source where the user evaluates those offers, and from there they prompt keplr to sign the messages.

(2) The wallet bridge is also used for other things, such as letting the dapp view all of the user's purses and current offers. However, this functionality could easily be included in a UI component directly inside the dapp, assuming we let the dapp connect to keplr to view the user's address.

### Problems with Wallet Bridge
It has combined dependencies on local storage, iframes, and CapTP. The implementation is complex and has resulted in many bugs and UX issues such as this one. Fixing this in its current form is technically difficult, error prone, and has maintenance risk as browser behaviors around these APIs may change.

### Proposed Solution
The ultimate goal here should be to remove the need for the wallet bridge entirely. Either through sign-mode-textual, metamask snaps, or upstream changes to keplr itself, we would like the user to be able to securely interact with dapps with nothing more than a browser extension. However, due to the problem (1) above presents, we require some trusted UI to evaluate offers.

Before then, we can radically simplify the wallet bridge implementation through eliminating (2) above. This means that the dapp would connect to Keplr directly, view the user's wallet address, and load all their purses and offers directly. This functionality could be encapsulated in a react/javascript component without requiring any type of bridge. This also brings us closer to an implementation congruent with the final goal of not requiring a bridge at all.

Doing the above will get the dapp to load all of the data and handle all user flows up to the point of proposing offers. We still need a wallet bridge to do that last bit, however it can be radically simplified now. The dapp simply needs a way to send offer proposals to the wallet UI and nothing more. This makes the bridge no longer a "live" connection to the wallet UI browser tab, but rather a component for dropping offer configs into local storage. We can remove the CapTP mechanisms and radically simplify the logic, and eliminate any bugs that resulted from multiple tabs at once or loss of connection.

### Caveats
- The dapp would need to specify IDs for proposed offers if it wants to track their statuses. I think for the PSM we could remove that feedback around offers to simplify things, the wallet itself is already good enough for tracking them.
- The dapp has to have some way to select the network when connecting to Keplr, right now the wallet bridge removes that concern
